mdn/src/main/java/org/apache/james/mdn/MDNReport.java

```diff
@@ -0,0 +1,135 @@
+            return this;
+        }
+
+        public Builder finalRecipientField(FinalRecipient finalRecipientField) {
+            this.finalRecipientField = Optional.of(finalRecipientField);
+            return this;
+        }
+
+        public Builder originalMessageIdField(OriginalMessageId originalMessageIdField) {
+            this.originalMessageIdField = Optional.of(originalMessageIdField);
+            return this;
+        }
+
+        public Builder dispositionField(Disposition dispositionField) {
+            this.dispositionField = Optional.of(dispositionField);
+            return this;
+        }
+
+        public MDNReport build() {
+            Preconditions.checkState(reportingUserAgentField.isPresent());
+            Preconditions.checkState(finalRecipientField.isPresent());
+            Preconditions.checkState(originalMessageIdField.isPresent());
+            Preconditions.checkState(dispositionField.isPresent());
+
+            return new MDNReport(reportingUserAgentField.get(),
+                originalRecipientField,
+                finalRecipientField.get(),
+                originalMessageIdField.get(),
+                dispositionField.get());
+        }
+
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static final String LINE_END = "\r\n";
+
+    private final ReportingUserAgent reportingUserAgentField;
+    private final Optional<OriginalRecipient> originalRecipientField;
+    private final FinalRecipient finalRecipientField;
+    private final OriginalMessageId originalMessageIdField;
+    private final Disposition dispositionField;
+
+    private MDNReport(ReportingUserAgent reportingUserAgentField, Optional<OriginalRecipient> originalRecipientField,
+                     FinalRecipient finalRecipientField, OriginalMessageId originalMessageIdField, Disposition dispositionField) {
+        this.reportingUserAgentField = reportingUserAgentField;
+        this.originalRecipientField = originalRecipientField;
+        this.finalRecipientField = finalRecipientField;
+        this.originalMessageIdField = originalMessageIdField;
+        this.dispositionField = dispositionField;
+    }
+
+    public ReportingUserAgent getReportingUserAgentField() {
+        return reportingUserAgentField;
+    }
+
+    public Optional<OriginalRecipient> getOriginalRecipientField() {
+        return originalRecipientField;
+    }
+
+    public FinalRecipient getFinalRecipientField() {
+        return finalRecipientField;
+    }
+
+    public OriginalMessageId getOriginalMessageIdField() {
+        return originalMessageIdField;
+    }
+
+    public Disposition getDispositionField() {
+        return dispositionField;
+    }
+
+    public String formattedValue() {
+        return reportingUserAgentField.formattedValue() + LINE_END
+            + originalRecipientField.map(value -> value.formattedValue() + "\r\n").orElse("")
```

Comment:
LINE_END ?

Reply:
... Done later... Sorry again

Reply:
Explained in your earlier (duplicated) comment